packages/frontend/src/components/FilterBar.tsx

```diff
@@ -14,13 +14,15 @@ export function FilterBar({ categories }: FilterBarProps) {
   );
 
   const submitFilter = useCallback(
-    (value: string) => {
-      setQuery(value);
+    (queryInput: string | null, filterByInput: string | null) => {
+      if (queryInput !== null) setQuery(queryInput);
+
+      if (filterByInput !== null) setFilterBy(filterByInput);
```

Comment:
Not a blocker, but I think we can ditch the `useState` for both `query` and `filterBy` - right now, we're storing them in / loading them from the URL, so we don't need to store the state twice (once in the URL, once in the React `state`).